sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/FormTrainingAsyncClientJavaDocCodeSnippets.java

```diff
@@ -44,39 +46,39 @@ public void beginTraining() {
                                 key, customFormModelField.getName(), customFormModelField.getAccuracy())));
                 });
             });
-        // END: com.azure.ai.formrecognizer.FormTrainingAsyncClient.beginTraining#string-boolean
+        // END: com.azure.ai.formrecognizer.training.FormTrainingAsyncClient.beginTraining#string-boolean
     }
 
     /**
-     * Code snippet for {@link FormTrainingAsyncClient#beginTraining} with options
+     * Code snippet for {@link FormTrainingAsyncClient#beginTraining(String, boolean, TrainModelOptions, Duration)}
+     * with options
      */
     public void beginTrainingWithOptions() {
-        // BEGIN: com.azure.ai.formrecognizer.FormTrainingAsyncClient.beginTraining#string-boolean-boolean-string-Duration
+        // BEGIN: com.azure.ai.formrecognizer.training.FormTrainingAsyncClient.beginTraining#string-boolean-trainModelOptions-Duration
         String trainingSetSource = "{training-set-SAS-URL}";
-        boolean isIncludeSubFolders = false; // {is-include-subfolders}
-        String filePrefix = "{file-prefix}";
-        boolean useLabelFile = true;
-
-        formTrainingAsyncClient.beginTraining(trainingSetSource, useLabelFile, isIncludeSubFolders, filePrefix,
-            Duration.ofSeconds(5)).subscribe(recognizePollingOperation -> {
-                // if training polling operation completed, retrieve the final result.
-                recognizePollingOperation.getFinalResult().subscribe(customFormModel -> {
-                    System.out.printf("Model Id: %s%n", customFormModel.getModelId());
-                    System.out.printf("Model Status: %s%n", customFormModel.getModelStatus());
-                    customFormModel.getSubModels().forEach(customFormSubModel ->
-                        customFormSubModel.getFieldMap().forEach((key, customFormModelField) ->
-                            System.out.printf("Form Type: %s Field Text: %s Field Accuracy: %s%n",
-                                key, customFormModelField.getName(), customFormModelField.getAccuracy())));
+        TrainModelOptions trainModelOptions = new TrainModelOptions().setIncludeSubFolders(false).setPrefix(
+            "Invoice");
+
+        formTrainingAsyncClient.beginTraining(trainingSetSource, true, trainModelOptions, Duration.ofSeconds(5)
+        ).subscribe(recognizePollingOperation -> {
```

Comment:
this seems odd, can be closed in last line and then begin .subscribe( from this line.